test/correctness/simd_op_check_arm.cpp

```diff
@@ -548,19 +548,31 @@ class SimdOpCheckARM : public SimdOpCheckTest {
                         // If we're widening the type by a factor of four
                         // as well as reducing by a factor of four, we
                         // expect vpaddl followed by vpadal
-                        check(arm32 ? "vpaddl.s8" : "saddlp", 8, sum_(i32(in_i8(f * x + r))));
-                        check(arm32 ? "vpaddl.u8" : "uaddlp", 8, sum_(i32(in_u8(f * x + r))));
-                        check(arm32 ? "vpaddl.u8" : "uaddlp", 8, sum_(u32(in_u8(f * x + r))));
+                        if (target.has_feature(Target::ARMDotProd)) {
```

Comment:
Were we never testing with this feature before?

Reply:
Nope!

Reply:
Yikes. Great catch!